homeassistant/components/ozw/entity.py

```diff
@@ -301,3 +306,171 @@ def create_value_id(value: OZWValue):
+                payload = new_value.lower() == "true"
+                value.send_value(payload)
+                return OZWValidationResponse.process_success(payload)
+
+            return OZWValidationResponse.process_fail(
+                ERR_NOT_SUPPORTED,
+                "Configuration parameter requires true of false",
+            )
+
+        return OZWValidationResponse.process_fail_on_type(value, new_value)
+
+    # List value can be passed in as string or int
+    if value.type == ValueType.LIST:
+        try:
+            new_value = int(new_value)
+        except ValueError:
+            pass
+        if not isinstance(new_value, str) and not isinstance(new_value, int):
+            return OZWValidationResponse.process_fail_on_type(value, new_value)
+
+        for option in value.value["List"]:
+            if new_value not in (option["Label"], option["Value"]):
+                continue
+            payload = int(option["Value"])
+            value.send_value(payload)
+            return OZWValidationResponse.process_success(payload)
+
+        return OZWValidationResponse.process_fail(
+            ERR_NOT_SUPPORTED,
+            "Invalid value %s for parameter %s",
+            new_value,
+            parameter_index,
+        )
+
+    if value.type == ValueType.STRING:
+        if not isinstance(new_value, str):
+            return OZWValidationResponse.process_fail_on_type(value, new_value)
+        value.send_value(new_value)
+        return OZWValidationResponse.process_success(new_value)
+
+    if value.type in (ValueType.INT, ValueType.BYTE, ValueType.SHORT):
+        try:
+            new_value = int(new_value)
+        except ValueError:
+            return OZWValidationResponse.process_fail_on_type(value, new_value)
+        if new_value > value.max or new_value < value.min:
+            return OZWValidationResponse.process_fail(
+                ERR_NOT_SUPPORTED,
+                "Value %s out of range for parameter %s (Min: %s Max: %s)",
+                new_value,
+                parameter_index,
+                value.min,
+                value.max,
+            )
+        value.send_value(new_value)
+        return OZWValidationResponse.process_success(new_value)
+
+    return OZWValidationResponse.process_fail(
+        ERR_NOT_SUPPORTED,
+        "Value type of %s for parameter %s not supported",
+        str(value.type),
+        parameter_index,
+    )
+
+
+def get_config_parameters(node, node_instance_id=None):
+    """Get config parameter from a node."""
+    command_class = node.get_command_class(CommandClass.CONFIGURATION, node_instance_id)
+    if not command_class:
+        return OZWValidationResponse.process_fail(
+            ERR_NOT_FOUND, "Configuration parameters for OZW Node Instance not found"
+        )
+
+    values = []
+
+    for value in command_class.values():
+        value_to_return = {}
+        if value.read_only or value.type == ValueType.BUTTON:
```

Comment:
thanks @kpine you're right, my main goal here was to follow the original PR, but I can add a comment in the code. I will also remove String and make it one of the unavailable options, as well as Decimal and Raw.

I think it makes sense to skip BitSet for now since OZW doesn't recognize it, and to add it in later

Reply:
Just to add, it doesn't look like `python-openzwave-mqtt` supports Decimal or Raw so those would default to `UNKNOWN` I'm guessing which I am also excluding

Reply:
Decimal is defined [here](https://github.com/cgarwood/python-openzwave-mqtt/blob/465aa53cf5b460a4d4a88fa99814f3fde47aafcb/openzwavemqtt/const.py#L79). I submitted an [issue](https://github.com/cgarwood/python-openzwave-mqtt/issues/90) for the other types. BitSet is supported by ozwdaemon, Button is broken.

Just another idea to throw out there. OZW has a "config" genre type. Unfortunately, python-openzwave-mqtt does not expose it currently. If it did, you could just iterate over all of the values and skip the ones that aren't the "config" genre. This would remove the need to know about command classes, and you could remove the node instance from the API entirely. BTW, OZW only supports config params for instance 1.

Reply:
good to know. I will add `Decimal` for now and remove the instance input since it is unnecessary. The other types can be added later (or the switch to use config genre) to minimize the number of changes this PR is creating